docs/transpile/custom-transpiler-pass.ipynb

```diff
@@ -0,0 +1,239 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Transpiler Passes\n",
+    "\n",
+    "Transpiler passes are classified into one of two classes: Analysis passes and Transformation passes. Passes in general work with the Directed Acyclic Graph (DAG) **TODO: insert link here** and the `property_set`, a dictionary-like object for storing properties determined by analysis passes. Analysis passes work with the DAG, but are not allowed to modify it. This contrasts with transformation passes, which do modify the DAG, and can read (but not write to) `property_set`. "
```

Comment:
Can you expand this intro to discuss what is going to be covered in this page more explicitly? Like "In this page we're going to explain what you need to know to create your own traspiler pass, looking at a specific example of pauli twirling..." or something similar?

Reply:
Also, for the first line I feel like I (as the reader) am missing some context about why you're telling me about the analysis and transformation passes. Can you add an extra line about why I should care? e.g. "when writing your own transpiler pass you should know that there are 2 classes of passes..."